### PR TITLE
Update help feature

### LINKED ISSUE #52: Update HelpCommand.java

### PR DESCRIPTION
**Pull request by huangtian666**
Fixes #52

### CODE REVIEW

File: src/test/java/seedu/address/testutil/JavaFxInitializer.java [hunk 1]
```diff
@@ -20,6 +20,7 @@ public class JavaFxInitializer {
      * </p>
      */
     public static synchronized void init() {
+
```
Comment (jeronkk) [inline src/test/java/seedu/address/testutil/JavaFxInitializer.java]: I think you do not need to commit this change.

File: src/main/resources/view/HelpWindow.css [hunk 2]
```diff
@@ -87,4 +87,4 @@
 /* Label styles */
 #helpMessage {
     -fx-text-fill: white;
-}
```
Comment (jeronkk) [inline src/main/resources/view/HelpWindow.css]: This change does not require an individual commit.

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 3]
```diff
@@ -48,7 +48,11 @@ public ListCommand(Predicate<Person> predicate, String successMessage, String ke
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
+
+        assert predicate != null : "Predicate should be initialized";
```
Comment (jeronkk) [inline src/main/java/seedu/address/logic/commands/ListCommand.java:52]: Consider replacing this with Objects.requireNonNull(predicate, "Predicate should be initialized"); for more consistent null checking. assert statements may be disabled in production, which could allow a null to slip through unnoticed.

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 4]
```diff
@@ -48,7 +48,11 @@ public ListCommand(Predicate<Person> predicate, String successMessage, String ke
...
         model.updateFilteredPersonList(this.predicate);
+
+        assert model.getFilteredPersonList() != null : "Filtered list should not be null";
```
Comment (jeronkk) [inline src/main/java/seedu/address/logic/commands/ListCommand.java:55]: Do we expect getFilteredPersonList() to ever return null after updateFilteredPersonList(...)? If this can't be null under normal conditions, this assertion might be unnecessary or better handled with a unit test.

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 5]
```diff
@@ -48,7 +48,11 @@ public ListCommand(Predicate<Person> predicate, String successMessage, String ke
     @Override
     public CommandResult execute(Model model) {
```
Comment (jeronkk) [inline src/main/java/seedu/address/logic/commands/ListCommand.java:49]: Consider grouping all your precondition checks (e.g., requireNonNull, assert) at the very start of the method. This can make your command’s preconditions easier to read and maintain.

File: src/main/java/seedu/address/logic/commands/ListCommand.java
Comment (jeronkk) [inline src/main/java/seedu/address/logic/commands/ListCommand.java:1]: The new assertions are currently not covered by tests, which is causing the Codecov patch check to fail. You could add a test case that executes this command with a valid predicate and model to ensure those lines are exercised.

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 6]
```diff
@@ -48,7 +48,11 @@ public ListCommand(Predicate<Person> predicate, String successMessage, String ke
...
         requireNonNull(model);
+
+        assert predicate != null : "Predicate should be initialized";
         model.updateFilteredPersonList(this.predicate);
+
+        assert model.getFilteredPersonList() != null : "Filtered list should not be null";
         if (model.getFilteredPersonList().isEmpty()) {
```
Comment (jeronkk) [inline src/main/java/seedu/address/logic/commands/ListCommand.java:56]: As this command grows in logic, you might consider extracting the filtered list logic or message generation into a helper method. For now it's fine, but worth keeping in mind for readability.

**Review by jeronkk — APPROVED** [review]
Just some comments to take note of, please take a look at the suggestions and make the changes if necessary. Great work overall!

File: src/test/java/seedu/address/ui/CommandUsageInfoTest.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    public void record_immutableProperties() {
+        // Once a record is created, its fields cannot be changed (immutability).
+        CommandUsageInfo info = new CommandUsageInfo("delete", "delete INDEX", "delete 3");
+
+        // We can only read the values, not modify them
+        assertEquals("delete", info.command());
+        assertEquals("delete INDEX", info.format());
+        assertEquals("delete 3", info.example());
+    }
+}
```
Comment (brvndonkoh) [inline src/test/java/seedu/address/ui/CommandUsageInfoTest.java:35]: Great set of tests for the `CommandUsageInfo` record! Clear, simple, and effective. You've covered both correct value assignment and immutability. Just a small suggestion to consider negative test cases if relevant. Excellent work Huang Tian!

File: src/test/java/seedu/address/ui/CommandUsageInfoTest.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Test
+    public void record_immutableProperties() {
+        // Once a record is created, its fields cannot be changed (immutability).
+        CommandUsageInfo info = new CommandUsageInfo("delete", "delete INDEX", "delete 3");
+
+        // We can only read the values, not modify them
+        assertEquals("delete", info.command());
+        assertEquals("delete INDEX", info.format());
+        assertEquals("delete 3", info.example());
```
Comment (brvndonkoh) [inline src/test/java/seedu/address/ui/CommandUsageInfoTest.java:33]: Add a negative test case (if possible) to ensure the constructor doesn't accept null or invalid values, depending on whether `CommandUsageInfo` has any such constraints. This would improve robustness.

File: src/test/java/seedu/address/ui/CommandConfigTest.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+public class CommandConfigTest {
+
+    @Test
+    public void getAllCommands_returnsNonNullList() {
+        List<CommandUsageInfo> commands = CommandConfig.getAllCommands();
+        assertNotNull(commands, "Command list should not be null");
```
Comment (brvndonkoh) [inline src/test/java/seedu/address/ui/CommandConfigTest.java:15]: Excellent use of `assertNotNull.` This is a simple but essential check to confirm that the method doesn't return a null reference, which can prevent subtle bugs down the line.

**Review by brvndonkoh — APPROVED** [review]

File: src/main/java/seedu/address/logic/commands/ListCommand.java (hunk 3 above)
Comment (brvndonkoh) [reply src/main/java/seedu/address/logic/commands/ListCommand.java:52]: I agree! Replacing assert with Objects.requireNonNull(predicate, "Predicate should be initialized") is a good call for runtime safety.

File: src/main/java/seedu/address/logic/commands/ListCommand.java (hunk 5 above)
Comment (brvndonkoh) [reply src/main/java/seedu/address/logic/commands/ListCommand.java:49]: I agree with Jeron. Having them clearly grouped at the start definitely helps signal the method’s assumptions up front and makes the code easier to read and maintain.

**Review by brvndonkoh — APPROVED** [review]
